internal/updater/updater.go

```diff
@@ -72,36 +73,54 @@ func NewAvailableUpdate(channel, version, platform, path, sha256, tag string) *A
 	}
 }
 
+func NewAvailableUpdateFromGraph(au *graph.AvailableUpdate) *AvailableUpdate {
+	if au == nil {
+		return &AvailableUpdate{}
+	}
+	return NewAvailableUpdate(au.Channel, au.Version, au.Platform, au.Path, au.Sha256, "")
+}
+
+func (u *AvailableUpdate) IsValid() bool {
+	return u != nil && u.Channel != "" && u.Version != "" && u.Platform != "" && u.Path != "" && u.Sha256 != ""
+}
+
+func (u *AvailableUpdate) Equals(origin *Origin) bool {
+	return u.Channel == origin.Channel && u.Version == origin.Version
+}
+
 type Update struct {
-	AvUpdate *AvailableUpdate
-	Origin   *Origin
-	url      string
-	tmpDir   string
-	an       analytics.Dispatcher
+	AvailableUpdate *AvailableUpdate
+	Origin          *Origin
+
+	url    string
+	tmpDir string
+	an     analytics.Dispatcher
 }
 
-func NewUpdate(an analytics.Dispatcher, origin *Origin, avUpdate *AvailableUpdate) *Update {
+// NewUpdateByOrigin returns an instance of Update. Allowing origin to be set
+// is useful for testing.
+func NewUpdateByOrigin(an analytics.Dispatcher, origin *Origin, avUpdate *AvailableUpdate) *Update {
 	apiUpdateURL := constants.APIUpdateURL
 	if url, ok := os.LookupEnv("_TEST_UPDATE_URL"); ok {
 		apiUpdateURL = url
 	}
 
 	return &Update{
-		AvUpdate: avUpdate,
-		Origin:   origin,
-		url:      apiUpdateURL + "/" + avUpdate.Path,
-		an:       an,
+		AvailableUpdate: avUpdate,
+		Origin:          origin,
+		url:             apiUpdateURL + "/" + avUpdate.Path,
+		an:              an,
 	}
 }
 
-func NewUpdateCurrent(an analytics.Dispatcher, avUpdate *AvailableUpdate) *Update {
-	return NewUpdate(an, NewOriginDefault(), avUpdate)
+func NewUpdate(an analytics.Dispatcher, avUpdate *AvailableUpdate) *Update {
+	return NewUpdateByOrigin(an, NewOriginDefault(), avUpdate)
 }
 
-func (u *Update) ShouldSkip() bool {
-	return os.Getenv(constants.ForceUpdateEnvVarName) != "true" &&
-		u.AvUpdate.Channel == u.Origin.Channel &&
-		u.AvUpdate.Version == u.Origin.Version
+func (u *Update) NotNeeded() bool {
```

Comment:
Nit: Suggest naming this `IsValid()` as well. "Not" functions tend to lead to double negatives.